packages/quantic/force-app/solutionExamples/main/lwc/resultAttachToCase/resultAttachToCase.js

```diff
@@ -0,0 +1,165 @@
+import {api, LightningElement, track} from 'lwc';
+import {
+  registerComponentForInit,
+  initializeWithHeadless,
+  getHeadlessBundle,
+} from 'c/quanticHeadlessLoader';
+// @ts-ignore
+import {buildResultToAttach, buildResultToDetach} from 'c/attachToCaseUtils';
+import attachResult from '@salesforce/label/c.AttachResult';
+import detachResult from '@salesforce/label/c.DetachResult';
+import resultIsAttached from '@salesforce/label/c.ResultIsAttached';
+// @ts-ignore
+import {attachToCase, detachFromCase} from 'c/attachToCaseService';
+
+/** @typedef {import("coveo").AttachToCase} AttachToCase */
+/** @typedef {import("coveo").Result} Result */
+
+export default class ResultAttachToCase extends LightningElement {
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The result to attach/detach.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * The label displayed in the tooltip of the result action button.
+   * @api
+   * @type {string}
+   */
+  @api label;
+  /**
+   * Set to true so the resultAttachToCase behaves only as a visual indicator when results are attached.
+   * @api
+   * @type {boolean}
+   */
+  @api readOnly;
+
+  /** @type {string} */
+  eventName = 'insightpanel__attachtocase';
+
+  labels = {
+    attachResult: attachResult,
+    detachResult: detachResult,
+    resultIsAttached: resultIsAttached,
+  };
+
+  @track _isAttached;
+  @track _isLoading;
```

Comment:
Is the `@track` decorator needed to support older versions? Or something else?
https://developer.salesforce.com/docs/platform/lwc/guide/reference-decorators.html?_ga=2.260979085.1676241658.1749743693-1077754324.1749743693